BrainPortal/app/views/groups/show.html.erb

```diff
@@ -110,28 +113,65 @@
+        default_text += array_to_table(group_editors.sort{ |a,b| a.login.casecmp(b.login)}, :table_class => 'simple bordered float_left', :cols => 20, :min_data => 20, :fill_by_columns => true ) { |u,r,c| link_to_user_with_tooltip(u)}
+      end
+      if group_viewers.any? then
+        default_text += '<label>Cannot edit:</label>'
+        default_text += array_to_table(group_viewers.sort{ |a,b| a.login.casecmp(b.login)}, :table_class => 'simple bordered float_left', :cols => 20, :min_data => 20, :fill_by_columns => true ) { |u,r,c| link_to_user_with_tooltip(u)}
+      end
+      default_text = default_text.html_safe
+    else
+      members_header = 'Members'
+      default_text = array_to_table(group_members.sort{ |a,b| a.login.casecmp(b.login)}, :table_class => 'simple bordered float_left', :cols => 20, :min_data => 20, :fill_by_columns => true ) { |u,r,c| link_to_user_with_tooltip(u)}
+    end
 
-    <% default_text = array_to_table(group_members.sort{ |a,b| a.login.casecmp(b.login)}, :table_class => 'simple bordered float_left', :cols => 20, :min_data => 20, :fill_by_columns => true ) { |u,r,c| link_to_user_with_tooltip(u) } %>
+  %>
 
+  <%= show_table(@group, :as => :group, :header => members_header, :edit_condition => @group.can_be_edited_by?(current_user) && (!current_user.has_role?(:normal_user) || group_members.count > 1 || open_invites.count > 0 )) do |t| %>
     <% t.edit_cell(:user_id, :show_width => 2,
                              :no_header  => 'Members',
                              :content    => default_text) do %>
-      <% if current_user.has_role? :normal_user %>
+      <% if current_user.has_role? :normal_user or @group.is_a?(WorkGroup) %>
+        <table class="button_table">
+          #todo personally I like more the look of access_profile_label style than button table, to discuss
+          <tr>
+            <th>Username</th>
+            <% if @group.is_a?(WorkGroup) %>
+              <th>Can edit?</th>
+            <% end %>
+            <th>Revoke membership</th>
+          </tr>
         <% group_members.each do |u| %>
-          <%= link_to_user_with_tooltip u %><%= link_to(": Remove", group_path(@group, "group[user_ids]" => (@group.user_ids - [u.id]), update_users: true ), :method => :put, :class => "action_link") unless u == current_user %>
-          <br>
+          <tr>
+            <td>
+              <%= link_to_user_with_tooltip u %>
+            </td>
+            <%  if @group.is_a?(WorkGroup) %>
+              <td>
+                <%= check_box_tag("group[editor_ids][]", u.id, @group.editor_ids.include?(u.id),
+                      :checked => u.id == @group.creator.id, :disabled=> u.id == @group.creator.id ) %>
+              </td>
+            <% end %>
+            <td>
+              <%= link_to("Remove", group_path(@group, "group[user_ids]" => (@group.user_ids - [u.id]), update_users: true ), :method => :put, :class => "action_link") unless u == current_user %>
```

Comment:
Please don't do that kind of POST, when we have 1000+ users you get 1000 links each with 999 user ids in them, it's ridiculous

Reply:
@prioux That's Tarek's code. I can refactor that of course.

Reply:
Yes that's just awful. Have a look at what the HTMl looks like in production using your admin account, I bet it's horrific.